test/unit-test/test_btc_sign.c

```diff
@@ -166,6 +166,76 @@ static void _test_btc_sign_init(void** state)
         invalid.coin = _BTCCoin_MAX + 1;
         assert_int_equal(APP_BTC_ERR_INVALID_INPUT, app_btc_sign_init(&invalid, &next));
     }
+    { // test invalid account keypath
+        tst_app_btc_reset();
+        BTCSignInitRequest invalid = init_req_valid;
+        invalid.script_configs[0].keypath[2] = BIP32_INITIAL_HARDENED_CHILD + 100;
+        assert_int_equal(APP_BTC_ERR_INVALID_INPUT, app_btc_sign_init(&invalid, &next));
+    }
+    { // no script configs is invalid
+        tst_app_btc_reset();
+        BTCSignInitRequest invalid = init_req_valid;
+        invalid.script_configs_count = 0;
+        assert_int_equal(APP_BTC_ERR_INVALID_INPUT, app_btc_sign_init(&invalid, &next));
+    }
+    { // can't mix simple type (singlesig) and multisig configs in one tx
+        tst_app_btc_reset();
+        BTCSignInitRequest invalid = init_req_valid;
+        invalid.script_configs_count = 2;
+        BTCScriptConfigWithKeypath sc = {
+            .script_config =
+                {
+                    .which_config = BTCScriptConfig_simple_type_tag,
+                    .config =
+                        {
+                            .simple_type = BTCScriptConfig_SimpleType_P2WPKH_P2SH,
+                        },
+                },
+            .keypath_count = 3,
+            .keypath =
+                {
+                    49 + BIP32_INITIAL_HARDENED_CHILD,
+                    0 + BIP32_INITIAL_HARDENED_CHILD,
+                    0 + BIP32_INITIAL_HARDENED_CHILD,
+                },
+        };
+        invalid.script_configs[1] = sc;
+
+        assert_int_equal(APP_BTC_OK, app_btc_sign_init(&invalid, &next));
+        tst_app_btc_reset();
+        invalid.script_configs[0].keypath[2]++;
+        assert_int_equal(APP_BTC_ERR_INVALID_INPUT, app_btc_sign_init(&invalid, &next));
+    }
+    { // can't mix script configs from different bip44 accounts
+        // (mixing input scripts is allowed, but only from the same bip44 account).
```

Comment:
I think this comment refers to the previous test and the other way around: this one is about multisig and the previous is about different accounts. So, swap them.

Reply:
very good catch, fixed